File: bestia-models/src/test/java/net/bestia/model/map/WalkspeedTest.java
package net.bestia.model.map;

import org.junit.Assert;
import org.junit.Test;

public class WalkspeedTest {

    @Test(expected = IllegalArgumentException.class)
    public void fromInt_outOfRange_throws() {
        Walkspeed.Companion.fromInt(2003);
    }

    @Test(expected = IllegalArgumentException.class)
    public void fromInt_negative_throws() {
        Walkspeed.Companion.fromInt(-1);
    }

    @Test
    public void fromInt_ok() {
        Walkspeed.Companion.fromInt(100);
        Walkspeed.Companion.fromInt(0);
        Walkspeed.Companion.fromInt(300);
    }

    @Test(expected = IllegalArgumentException.class)
    public void fromFloat_outOfRange_throws() {
        new Walkspeed(3.7f);
    }

    @Test(expected = IllegalArgumentException.class)
    public void fromFloat_negative_throws() {
        new Walkspeed(-1.7f);
    }

    @Test
    public void fromFloat_ok() {
        new Walkspeed(1.7f);
        new Walkspeed(0f);
        new Walkspeed(Walkspeed.MAX_WALKSPEED);
    }

    @Test
    public void getSpeed_ok() {
        Walkspeed ws = new Walkspeed(Walkspeed.MAX_WALKSPEED);
        Assert.assertEquals(0.01f, Walkspeed.MAX_WALKSPEED, ws.getSpeed());
    }

    @Test
    public void toInt_ok() {
        Walkspeed ws = new Walkspeed(Walkspeed.MAX_WALKSPEED);
        Assert.assertEquals(Walkspeed.MAX_WALKSPEED_INT, ws.toInt());

        ws = new Walkspeed(0f);
        Assert.assertEquals(0, ws.toInt());

        ws = Walkspeed.Companion.fromInt(100);
        Assert.assertEquals(100, ws.toInt());
    }
}

File: bestia-models/src/test/kotlin/net/bestia/model/map/WalkspeedTest.kt
package net.bestia.model.map

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test

class WalkspeedTest {

  @Test
  fun fromInt_outOfRange_throws() {
    assertThrows(IllegalArgumentException::class.java) { Walkspeed.fromInt(2003) }
  }

  @Test
  fun fromInt_negative_throws() {
    assertThrows(IllegalArgumentException::class.java) {
      Walkspeed.fromInt(-1)
    }
  }

  @Test
  fun fromInt_ok() {
    Walkspeed.fromInt(100)
    Walkspeed.fromInt(0)
    Walkspeed.fromInt(300)
  }

  @Test
  fun fromFloat_outOfRange_throws() {
    assertThrows(IllegalArgumentException::class.java) {
      Walkspeed(3.7f)
    }
  }

  @Test
  fun fromFloat_negative_throws() {
    assertThrows(IllegalArgumentException::class.java) {
      Walkspeed(-1.7f)
    }
  }

  @Test
  fun fromFloat_ok() {
    Walkspeed(1.7f)
    Walkspeed(0f)
    Walkspeed(Walkspeed.MAX_WALKSPEED)
  }

  @Test
  fun getSpeed_ok() {
    val ws = Walkspeed(Walkspeed.MAX_WALKSPEED)
    assertEquals(0.01f, Walkspeed.MAX_WALKSPEED, ws.speed)
  }

  @Test
  fun toInt_ok() {
    var ws = Walkspeed(Walkspeed.MAX_WALKSPEED)
    assertEquals(Walkspeed.MAX_WALKSPEED_INT.toLong(), ws.toInt().toLong())

    ws = Walkspeed(0f)
    assertEquals(0, ws.toInt().toLong())

    ws = Walkspeed.fromInt(100)
    assertEquals(100, ws.toInt().toLong())
  }
}
